feature-libs/quote/components/summary/actions/quote-summary-actions.component.ts

```diff
@@ -217,6 +220,27 @@ export class QuoteSummaryActionsComponent
     return (quote.totalPrice.value || 0) >= requestThreshold;
   }
 
+  /**
+   * Click handler for download button.
+   *
+   * @param quoteCode - the quote ID (aka code)
+   * @param attachments - array of attachments belonnging to the quote
+   */
+  onDownloadAttachment(quoteCode: string, attachments: QuoteAttachment[]) {
+    // Extracting the attachment ID from the first entry in the array
+    // as there would be only one proposal document attached for each quote
+    const attachmentId = attachments[0].id;
+    const filename = attachments[0].filename || attachmentId;
+    this.quoteFacade
+      .downloadAttachment(quoteCode, attachmentId)
+      .subscribe((res) => {
+        const url = window.URL.createObjectURL(
+          new Blob([res], { type: res.type })
+        );
+        this.fileDownloadService.download(url, `${filename}.pdf`);
```

Comment:
Please cover this by tests

Reply:
I'm already checking if the download service is being called in `quote-summary-actions.component.spec.ts` line no 1116:
` expect(fileDownloadService.download).toHaveBeenCalled();
`

Reply:
The test is not properly executed. Please check the test output when running `npx nx run quote:test --codeCoverage=true `.
You will also see that line 
 `this.fileDownloadService.download(url, `${filename}.pdf`);`
 is not covered by tests.
 
![Screenshot 2024-05-16 at 16 12 51](https://github.com/SAP/spartacus/assets/33626130/e3cf3ad1-7e79-4c02-9048-06f73a1c11d6)

Reply:
Updated the unit test to cover the line.